src/lib/operators/jit_operator/operators/jit_abstract_operator.hpp

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <vector>
+
+#include "operators/jit_operator/jit_types.hpp"
+
+namespace opossum {
+
+/**
+ * JitAbstractOperator is the abstract super class of all operators used within a JitOperator.
+ * Usually, multiple operators are linked together to form an operator chain.
+ * The operators work in a push-based fashion: The virtual "next" function is called for each tuple.
+ * The operator can then process the tuple and finally call its own "emit" function to pass the tuple
+ * on to the next operator in the chain.
+ */
+class JitAbstractOperator {
+ public:
+  using Ptr = std::shared_ptr<JitAbstractOperator>;
```

Comment:
We had that discussion some months ago, but currently, we don't have alias for shared pointers.